### PR TITLE
Improve diagnostics of incompatible attachments

### PR DESCRIPTION
**Pull request by scoopr**
**Description**
Attempt to improve the diagnostics of incompatible attachments

### CODE REVIEW

**Comment by scoopr** [comment]
The original error message got me confused for way too long, as I thought my color format or sample count was mismatching, but in the end it was the depth-stencil attachment that I had in the renderpass but didn't need for the pipeline, so the None didn't match with the depth format.

File: wgpu-core/src/command/bundle.rs [hunk 1]
```diff
@@ -719,9 +719,11 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
                             .use_extend(&*pipeline_guard, pipeline_id, (), ())
                             .unwrap();
 
-                        if !bundle_encoder.context.compatible(&pipeline.pass_context) {
-                            Err(RenderCommandError::IncompatiblePipeline)?
-                        }
+                        bundle_encoder
+                            .context
+                            .check_compatible(&pipeline.pass_context)
+                            .map_err(|e| RenderCommandError::IncompatiblePipeline(e))?;
```
Comment (kvark) [inline wgpu-core/src/command/bundle.rs:725]: nit: `map_err(RenderCommandError::IncompatiblePipeline)`

File: wgpu-core/src/command/render.rs [hunk 2]
```diff
@@ -1016,9 +1016,10 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
                         .use_extend(&*pipeline_guard, pipeline_id, (), ())
                         .unwrap();
 
-                    if !context.compatible(&pipeline.pass_context) {
-                        return Err(RenderCommandError::IncompatiblePipeline.into());
-                    }
+                    context
+                        .check_compatible(&pipeline.pass_context)
+                        .map_err(|e| RenderCommandError::IncompatiblePipeline(e))?;
```
Comment (kvark) [inline wgpu-core/src/command/render.rs:1021]: same here

File: wgpu-core/src/command/render.rs [hunk 3]
```diff
@@ -1512,9 +1513,9 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
                         .use_extend(&*bundle_guard, bundle_id, (), ())
                         .unwrap();
 
-                    if !context.compatible(&bundle.context) {
-                        return Err(RenderPassError::IncompatibleRenderBundle);
-                    }
+                    context
+                        .check_compatible(&bundle.context)
+                        .map_err(|e| RenderPassError::IncompatibleRenderBundle(e))?;
```
Comment (kvark) [inline wgpu-core/src/command/render.rs:1518]: and here

**Pull request merged**